gender_analysis/document.py

```diff
@@ -256,6 +258,9 @@ def _load_document_text(self):
             )
             raise FileNotFoundError(err) from original_err
 
+        # Remove Gutenberg header and footer.
+        if "project gutenberg" in text.lower():
```

Comment:
I definitely think the speed issue is coming from the `lower()` call here. I know we talked about in the hack session about how it's probably fine, but it looks like that's what's using up all of the resources. 

One thing you might consider is limiting the number of characters that it tries to check? I believe `split()` will make it such that it only operates on however many characters you want to check, so that you don't need to be lowercasing tens of thousands of characters at once.

Alternatively we could also make it such that the user manually selects if its a gutenberg document, like we talked about earlier, although that might be a bit of a hassle at this point.

Reply:
Hm. I tried this - in the latest commit, the _gutenberg_cleaner function only `lower()`s the first 100 characters. It doesn't seem to have had much of a performance impact. 

I am starting to suspect that the simple_cleaner function itself is what's holding things up, particularly since it needs to run about a hundred times every time we spin up a corpus for testing. Every solution that I've found online has used a collection of regular expressions to find what to strip out; I worry that that's what's taking so long.